src/csharp/Grpc.IntegrationTesting/XdsInteropClient.cs

```diff
@@ -0,0 +1,287 @@
+
+            [Option("qps", Default = 1)]
+            public int Qps { get; set; }
+
+            [Option("server", Default = "localhost:8080")]
+            public string Server { get; set; }
+
+            [Option("stats_port", Default = 8081)]
+            public int StatsPort { get; set; }
+
+            [Option("rpc_timeout_sec", Default = 30)]
+            public int RpcTimeoutSec { get; set; }
+
+            [Option("print_response", Default = false)]
+            public bool PrintResponse { get; set; }
+        }
+
+        ClientOptions options;
+
+        StatsWatcher statsWatcher = new StatsWatcher();
+
+        // make watcher accessible by tests
+        internal StatsWatcher StatsWatcher => statsWatcher;
+
+        internal XdsInteropClient(ClientOptions options)
+        {
+            this.options = options;
+        }
+
+        public static void Run(string[] args)
+        {
+            GrpcEnvironment.SetLogger(new ConsoleLogger());
+            var parserResult = Parser.Default.ParseArguments<ClientOptions>(args)
+                .WithNotParsed(errors => Environment.Exit(1))
+                .WithParsed(options =>
+                {
+                    var xdsInteropClient = new XdsInteropClient(options);
+                    xdsInteropClient.RunAsync().Wait();
+                });
+        }
+
+        private async Task RunAsync()
+        {
+            var server = new Server
+            {
+                Services = { LoadBalancerStatsService.BindService(new LoadBalancerStatsServiceImpl(statsWatcher)) }
+            };
+
+            string host = "0.0.0.0";
+            server.Ports.Add(host, options.StatsPort, ServerCredentials.Insecure);
+            Console.WriteLine($"Running server on {host}:{options.StatsPort}");
+            server.Start();
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            await RunChannelsAsync(cancellationTokenSource.Token);
+
+            await server.ShutdownAsync();
+        }
+
+        // method made internal to make it runnable by tests
+        internal async Task RunChannelsAsync(CancellationToken cancellationToken)
+        {
+            var channelTasks = new List<Task>();
+            for (int channelId = 0; channelId < options.NumChannels; channelId++)
+            {
+                var channelTask = RunSingleChannelAsync(channelId, cancellationToken);
+                channelTasks.Add(channelTask);
+            }
+
+            for (int channelId = 0; channelId < options.NumChannels; channelId++)
+            {
+                await channelTasks[channelId];
+            }
+        }
+
+        private async Task RunSingleChannelAsync(int channelId, CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"Starting channel {channelId}");
+            var channel = new Channel(options.Server, ChannelCredentials.Insecure);
+            var client = new TestService.TestServiceClient(channel);
```

Comment:
looks like we are creating channels in a way that will facilitate subchannel sharing - I'm not sure this is what we want though?

Especially as `options.NumChannels` becomes large, the behavior and performance of this test with and without subchannel sharing might vary somewhat dramatically (especially e.g. if the backends are setting `htttp2 max concurrent streams`).

Suggest either disabling subchannel sharing (using this arg:
https://github.com/grpc/grpc/blob/2d63189237a87a1402517ab0f5ac306c924072e6/test/core/tsi/alts/handshaker/alts_concurrent_connectivity_test.cc#L87), or being explicit about whether or not this test uses subchannel sharing by adding one more option to the test runner to control whether or not we will use subchannel sharing.